pkg/operator/ceph/cluster/status/conditions.go

```diff
@@ -0,0 +1,300 @@
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionInitialize(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnecting,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnected,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionFailure,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionIgnored,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionUpgrading,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionExpanding,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionExpanding(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionExpanding,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+// applying the latest condition on to the cluster CRD file
+func applyCondition(context *clusterd.Context, conditions *[]cephv1.RookConditions, newCondition cephv1.RookConditions) {
+	if name == "" {
+		return
+	} else {
+		cluster, err := context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("failed to get cluster from namespace %s prior to updating its condition to %s. %+v", namespace, newCondition.Type, err)
+		}
+		*conditions = append(*conditions, newCondition)
+		cluster.Status.Condition = *conditions
+		cluster.Status.FinalCondition = newCondition.Type
+		cluster.Status.Message = newCondition.Message
+		if newCondition.Status == v1.ConditionTrue {
+			logger.Infof("CephCluster %s status: %s. %s", namespace, cluster.Status.FinalCondition, cluster.Status.Message)
+		}
+		if _, err := context.RookClientset.CephV1().CephClusters(namespace).Update(cluster); err != nil {
+			logger.Errorf("failed to update cluster %s condition: %+v", namespace, err)
```

Comment:
Don't you want to return if you get an error?

Reply:
Yes. That makes sense.

Reply:
I guess just displaying it into the logger is enough.